libsoftwarecontainer/component-test/devicenodegateway_componenttest.cpp

```diff
@@ -97,3 +97,60 @@ TEST_F(DeviceNodeGatewayTest, TestOverwriteDeviceFails) {
     checkMode.wait();
     ASSERT_TRUE(checkMode.isSuccess());
 }
+
+struct testSetup
+{
+    const std::string firstConfiguration;
+    const std::string secondConfiguration;
```

Comment:
It would be nice to be able to use more than two configuration stages in the tests, since the implementation is not limited to two afaict.